This logic mistakes "clang" for "CL.EXE". Let's just cut out the Windows
stuff entirely.
Index: GCC/CMakeLists.txt
===================================================================
--- GCC/CMakeLists.txt.orig
+++ GCC/CMakeLists.txt
@@ -23,18 +23,6 @@ IF(APPLE)
 ENDIF(APPLE)
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_ANSI_CFLAGS}")
 
-IF(CMAKE_C_COMPILER MATCHES "cl")
-  STRING(REGEX REPLACE "/W([0-4])?" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-  SET(CMAKE_CXX_WARNING_LEVEL 0)
-  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W0")
-ELSE(CMAKE_C_COMPILER MATCHES "cl")
-  IF(CMAKE_C_COMPILER MATCHES "bcc32")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -w-")
-  ELSE(CMAKE_C_COMPILER MATCHES "bcc32")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -w")
-  ENDIF(CMAKE_C_COMPILER MATCHES "bcc32")
-ENDIF(CMAKE_C_COMPILER MATCHES "cl")
-
 ADD_DEFINITIONS(-DALMOST_STDC)
 
 SUBDIRS(config_cmake libiberty libdecnumber libcpp gcc)
